### PR TITLE
fix conformance for PCA, KMeans and KMeans init constructors

### PR DESCRIPTION
**Pull request by mchernov-intel**

### CODE REVIEW

**Review by rlnx — COMMENTED** [review]
I think examples should remain unchanged

File: cpp/oneapi/dal/algo/pca/common.hpp [hunk 1]
```diff
@@ -70,6 +70,10 @@ class descriptor : public descriptor_base<Task> {
     using float_t = Float;
     using method_t = Method;
 
+    explicit descriptor(std::int64_t component_count = 0) {
+        set_component_count(component_count);
+    }
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/pca/common.hpp:75]: Is it correct? At least it's useless to run PCA with zero components

File: examples/oneapi/cpp/source/kmeans/kmeans_lloyd_dense_batch.cpp [hunk 2]
```diff
@@ -33,8 +33,9 @@ int main(int argc, char const *argv[]) {
     const auto x_test = dal::read<dal::table>(dal::csv::data_source{test_data_file_name});
     const auto y_test = dal::read<dal::table>(dal::csv::data_source{test_label_file_name});
 
-    const auto kmeans_desc = dal::kmeans::descriptor<>()
-                                 .set_cluster_count(20)
+        const auto kmeans333_desc = dal::kmeans::descriptor<>();
```
Comment (KulikovNikita) [inline examples/oneapi/cpp/source/kmeans/kmeans_lloyd_dense_batch.cpp:36]: Please check indent

File: examples/oneapi/cpp/source/pca/pca_dense_batch.cpp [hunk 3]
```diff
@@ -23,8 +23,7 @@ using namespace oneapi;
 
 template <typename Method>
 void run(const dal::table& x_train, const std::string& method_name) {
-    const auto pca_desc = dal::pca::descriptor<float, Method>()
-        .set_component_count(5)
+    const auto pca_desc = dal::pca::descriptor<float, Method>(5)
```
Comment (KulikovNikita) [inline examples/oneapi/cpp/source/pca/pca_dense_batch.cpp:26]:
```suggestion
    const auto pca_desc = dal::pca::descriptor<float, Method>(/*component_count = */5)
```
I see lack of explicitness here. Maybe proposed code is better

**Review by KulikovNikita — CHANGES_REQUESTED** [review]

**Review by rlnx — APPROVED** [review]

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**